Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -1394,6 +1394,78 @@ void RenderBox::clearOverridingLogicalWidthLength()
         gOverridingLogicalWidthLengthMap->remove(this);
 }
 
+std::optional<FlowRelativeDirection> RenderBox::physicalToFlowRelativeDirectionMapping(PhysicalDirection direction, LayoutCalculationPhase layoutCalculationPhase) const
+{
+    // FIXME: Return type should be just FlowRelativeDirection once all code paths have been implemented
+    auto isHorizontalWritingMode = style().isHorizontalWritingMode();
+    auto isLeftToRightDirection = style().isLeftToRightDirection();
+    // vertical-rl and horizontal-bt writing modes
+    auto isFlippedBlocksWritingMode = style().isFlippedBlocksWritingMode();
+    if (layoutCalculationPhase == LayoutCalculationPhase::Positioning) {
+        if (isHorizontalWritingMode == containingBlock()->style().isHorizontalWritingMode()) {
+            switch (direction) {
+            case PhysicalDirection::Top:
+                return isHorizontalWritingMode ? FlowRelativeDirection::BlockStart : (isLeftToRightDirection ? FlowRelativeDirection::InlineStart : FlowRelativeDirection::InlineEnd);
```

Comment:
/* Unofficial Review */
Inner ternary operators are hard to follow. I suggest you break it down with some if statements.
```
if (isHorizontalWritingMode)
    return FlowRelativeDirection::BlockStart;
return isLeftToRightDirection ? FlowRelativeDirection::InlineStart : FlowRelativeDirection::InlineEnd;
```

Reply:
Changed